pilot/pkg/config/kube/gateway/testdata/eastwest-labelport.yaml

```diff
@@ -0,0 +1,57 @@
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: eastwestgateway
+  namespace: istio-system
+  labels:
+    topology.istio.io/network: "network-1"
+    networking.istio.io/gatewayPort: 35443
+spec:
+  gatewayClassName: istio
```

Comment:
for ambient we're going to need two different ways to represent SANs for verification during double HBONE (SAN verification for the gateway _and_ destination). the dest workload / svc in remote cluster's SAN can be derived from the service account on the workload entry.

another place we could leverage in theory is `subjectAltNames` on the service entry (if we took that approach as opposed to gateway API here). i am also open to using gateway API here so long as we leave a way to represent the expected SAN of the network gateway

i understand this is not a blocker outside of ambient, but i want to make sure what we go with here will also work for ambient

Reply:
"downstream" TLS is part of the spec: https://gateway-api.sigs.k8s.io/guides/tls/?h=tls

We can define an HBONE listener for the east-west gateway to be `proto: HTTP, tls: TERMINATE`. 

The only available kind for `certificateReferences` is `Secret`. This doesn't quite make sense for us. 

There is an [extensions](https://gateway-api.sigs.k8s.io/guides/tls/?h=tls#extensions) field we can use. 

> Gateway TLS configurations provides an options map to add additional TLS settings for implementation-specific features.

Reply:
Ideally we merge this PR first, and then add the capability for discovering [trusted gateway](https://docs.google.com/document/d/1MCRZWvlbZS6DLn5N4pD9ww2O-fL7jzi9-l0cWbqH1kk/edit#heading=h.4y7xkspbqkjh) identities along with the rest of the trusted gateway impl. 

Confident that there is a place in both our old Service API as well as somewhere we can at least (temporarily) use in GAMMA.